### PR TITLE
Downgrade qos to client sub if needed

### PR DESCRIPTION
**Pull request by GavinDmello**
Solution for #208

### CODE REVIEW

File: test/qos2.js [hunk 1]
```diff
@@ -209,6 +209,39 @@ test('subscribe QoS 0, but publish QoS 2', function (t) {
...
+    length: 14,
+    retain: false
+  }
+
+  subscribe(t, subscriber, 'hello', 1, function () {
+    subscriber.outStream.once('data', function (packet) {
+      delete packet.messageId
+      t.deepEqual(packet, expected, 'packet must match')
+      t.end()
+    })
```
Comment (mcollina) [inline test/qos2.js]: Can you add some checks on the follow up packets to verify that the full QoS flow is respected?

Comment (GavinDmello) [reply test/qos2.js]: Sure.

Comment (GavinDmello) [reply test/qos2.js]: @mcollina  I was just checking. You have helper functions in place to check the QoS flow.
 https://github.com/mcollina/aedes/blob/ca42eb0164e08538cabf2ddaa087ef1ff4b9dc8e/test/qos2.js#L11-L46 and https://github.com/mcollina/aedes/blob/ca42eb0164e08538cabf2ddaa087ef1ff4b9dc8e/test/helper.js#L77-L96

**Comment by mcollina** [comment]
Yes we do! Can you use them in this test?

**Comment by GavinDmello** [comment]
@mcollina I've already used the publish and subscribe helper functions in the test 👍.

**Pull request merged**